nilearn/experimental/surface/tests/test_surface_image.py

```diff
@@ -25,22 +26,23 @@ def test_compare_file_and_inmemory_mesh(mini_mesh, tmp_path):
     assert np.array_equal(left.faces, left_loaded.faces)
 
 
-def test_surface_image_shape(make_mini_img):
-    img = make_mini_img()
-    assert img.shape == (9,)
-    img = make_mini_img((3,))
-    assert img.shape == (3, 9)
-    img = make_mini_img((7, 3))
-    assert img.shape == (7, 3, 9)
```

Comment:
What I don't understand is why the data in SurfaceImage would ever be 3D.

AFAIU, the shape is supposed to be
 - `(1, n_vertices)` when there is only one image 
 - `(n_timepoints, n_vertices)` when we have timeseries surface image

Reply:
So @Remi-Gau mentioned 3D arrays might be possible with layer fMRI, where you can sample a given vertex at different cortical depths at a given time point.

However, the fact if we wanna support this still open for discussion

Reply:
ATM we're not supporting it. This should be discussed in a coredev meeting, I have no experience with layered fMRI on my side.

Reply:
My position on this would be: we were not supporting it, so let's not add it just yet.

I would revisit the issues when layer MRI user ask for it.